MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,386 @@
+    {
+        pcaConfig.knownAuthorities = @[pcaConfig.authority];
+    }
+    
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithConfiguration:pcaConfig error:&error];
+    
+    if (!application)
+    {
+        NSString *resultText = [NSString stringWithFormat:@"Failed to create PublicClientApplication:\n%@", error];
+        [self.resultView setString:resultText];
+        return;
+    }
+    
+    BOOL result = [application.tokenCache clearWithContext:nil error:&error];
+    
+    if (result)
+    {
+        [self.resultView setString:@"Successfully cleared cache."];
+        settings.currentAccount = nil;
+        
+        [[NSNotificationCenter defaultCenter] postNotificationName:MSALTestAppCacheChangeNotification object:self];
+    }
+    else
+    {
+        [self.resultView setString:[NSString stringWithFormat:@"Failed to clear cache, error = %@", error]];
+    }
+}
+
+- (IBAction)clearCookies:(id)sender
+{
+    NSHTTPCookieStorage* cookieStore = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    NSArray* cookies = cookieStore.cookies;
+    for (NSHTTPCookie* cookie in cookies)
+    {
+        [cookieStore deleteCookie:cookie];
+    }
+    
+    [_resultView setString:[NSString stringWithFormat:@"Cleared %lu cookies.", (unsigned long)cookies.count]];
+    
+    // Clear WKWebView cookies
+    if (@available(macOS 10.11, *)) {
+        NSSet *allTypes = [WKWebsiteDataStore allWebsiteDataTypes];
+        [[WKWebsiteDataStore defaultDataStore] removeDataOfTypes:allTypes
+                                                   modifiedSince:[NSDate dateWithTimeIntervalSince1970:0]
+                                               completionHandler:^{}];
+    }
+}
+
+- (IBAction)acquireTokenInteractive:(id)sender
+{
+    (void)sender;
+    
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    NSDictionary *currentProfile = [MSALTestAppSettings currentProfile];
+    NSString *clientId = [currentProfile objectForKey:MSAL_APP_CLIENT_ID];
+    NSString *redirectUri = [currentProfile objectForKey:MSAL_APP_REDIRECT_URI];
+    MSALAuthority *authority = [settings authority];
+    NSDictionary *extraQueryParameters = [NSDictionary msidDictionaryFromWWWFormURLEncodedString:[self.extraQueryParamsField stringValue]];
+    
+    NSError *error = nil;
+    
+    MSALPublicClientApplicationConfig *pcaConfig = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId
+                                                                                                   redirectUri:redirectUri
+                                                                                                     authority:authority];
+    if (self.validateAuthority.selectedSegment == 1)
+    {
+        pcaConfig.knownAuthorities = @[pcaConfig.authority];
+    }
+    
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithConfiguration:pcaConfig error:&error];
```

Comment:
nit: can we have common method to create MSALPublicClientApplication based on settings. It would help to avoid some duplicate code between AT and ATS calls.

Reply:
Sure. Done

Reply:
Also updated extraQueryParameter code to convert string to url encoded instead of expecting a url encoded string